test/integration/datasets/DatasetsRepository.test.ts

```diff
@@ -101,47 +190,68 @@ describe('DatasetsRepository', () => {
 
   describe('getDataset', () => {
     describe('by numeric id', () => {
+      let testDatasetIds: CreatedDatasetIdentifiers
+
+      beforeAll(async () => {
+        try {
+          testDatasetIds = await createDataset.execute(TestConstants.TEST_NEW_DATASET_DTO)
+        } catch (error) {
+          throw new Error('Tests beforeAll(): Error while creating test dataset')
+        }
+      })
+
+      afterAll(async () => {
+        try {
+          await deletePublishedDatasetViaApi(testDatasetIds.persistentId)
+        } catch (error) {
+          throw new Error('Tests afterAll(): Error while deleting test dataset')
+        }
```

Comment:
The try-catch block that handles the 'Error while deleting test dataset' appears to be duplicated in several places. Should we move this try-catch logic inside the `deletePublishedDatasetViaApi` function? That way, if any error occurs, it throws the 'Error while deleting test dataset' message, and we don’t need to repeat the logic everywhere

Reply:
I refactored the code to move the try catch _deletePublishedDatasetViaApi_ block, as well as other blocks, to the helper class methods.